webrender/src/device.rs

```diff
@@ -2749,15 +2749,17 @@ impl<B: hal::Backend> Device<B, hal::Graphics> {
         }
 
         // the blit caller code expects to be able to render to the target
-        if let Some(barrier) = dest_img.transit(
-            hal::image::Access::COLOR_ATTACHMENT_READ | hal::image::Access::COLOR_ATTACHMENT_WRITE,
-            hal::image::ImageLayout::ColorAttachmentOptimal,
-        ) {
-            cmd_buffer.pipeline_barrier(
-                PipelineStage::TOP_OF_PIPE .. PipelineStage::TRANSFER,
-                hal::memory::Dependencies::empty(),
-                &[barrier],
-            );
+        {
+            let mut barriers = Vec::new();
+            barriers.extend(src_img.transit(hal::image::Access::COLOR_ATTACHMENT_READ | hal::image::Access::COLOR_ATTACHMENT_WRITE, hal::image::ImageLayout::ColorAttachmentOptimal));
+            barriers.extend(dest_img.transit(hal::image::Access::COLOR_ATTACHMENT_READ | hal::image::Access::COLOR_ATTACHMENT_WRITE, hal::image::ImageLayout::ColorAttachmentOptimal));
```

Comment:
Do we need `COLOR_ATTACHMENT_READ` and `COLOR_ATTACHMENT_WRITE ` access bits for both images. Just adding `COLOR_ATTACHMENT_READ` to the `src_image` and `COLOR_ATTACHMENT_WRITE`  to the `dest_img` wouldn't be enough?

Reply:
Not sure, because in this PR i focused on the usage flag issue.
When i do the https://github.com/szeged/webrender/issues/99, i will double check these.